### ISSUE TITLE
Block incorrectly treated as unreachable when strict optional disabled

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6827]
This program doesn't generate an error when using `--no-strict-optional`, even though every line is reachable:

```py
from typing import NamedTuple, Dict

class C(NamedTuple):
    a: int
    b: str

x: Dict[int, C]
y = x.get(int())
if not y:
    1 + ''  # No error (considered unreachable)
```

Apparently mypy thinks that the tuple object is always truthy, even though it could also be `None` when strict optional checking is disabled.

**Comment by ilevkivskyi** [comment]
@JukkaL isn't this essentially a duplicate of https://github.com/python/mypy/issues/3601?

**Comment by JukkaL** [comment]
Yeah, looks like a duplicate.
